--- busted/outputHandlers/utfTerminal.lua
+++ busted/outputHandlers/utfTerminal.lua
@@ -14,10 +14,10 @@
   local busted = require 'busted'
   local handler = require 'busted.outputHandlers.base'()
 
-  local successDot = colors.green('?')
-  local failureDot = colors.red('?')
-  local errorDot   = colors.magenta('?')
-  local pendingDot = colors.yellow('?')
+  local successDot = colors.green('\183')
+  local failureDot = colors.red('\143')
+  local errorDot   = colors.magenta('\7')
+  local pendingDot = colors.yellow('\186')
 
   local pendingDescription = function(pending)
     local name = pending.name
