--- busted/context.lua
+++ busted/context.lua
@@ -14,10 +14,14 @@
 
 local function restore(state)
   setmetatable(_G, state.gmt)
-  for k,_ in next, _G, nil do
+  local names = {}
+  for k,_ in next, _G, nil do names[k] = true end
+  for k,_ in next, names, nil do
     rawset(_G, k, state.g[k])
   end
-  for k,_ in pairs(package.loaded) do
+  local names = {}
+  for k,_ in pairs(package.loaded) do names[k] = true end
+  for k,_ in pairs(names) do
     package.loaded[k] = state.loaded[k]
   end
 end
