--- lua/pl/path.lua
+++ lua/pl/path.lua
@@ -119,7 +119,7 @@
     return sub(s,i,i)
 end
 
-path.is_windows = utils.dir_separator == '\\'
+path.is_windows = false
 
 local other_sep
 -- !constant sep is the directory separator for this platform.
@@ -353,11 +353,11 @@
 function path.expanduser(P)
     assert_string(1,P)
     if at(P,1) == '~' then
-        local home = getenv('HOME')
-        if not home then -- has to be Windows
-            home = getenv 'USERPROFILE' or (getenv 'HOMEDRIVE' .. getenv 'HOMEPATH')
+        if sub(P,2, 2) == "/" then
+            return sub(P,2)
+        else
+            return "/" .. sub(P,2)
         end
-        return home..sub(P,2)
     else
         return P
     end
