--- lua/pl/stringx.lua
+++ lua/pl/stringx.lua
@@ -305,7 +305,7 @@
 
 local function _strip(s,left,right,chrs)
     if not chrs then
-        chrs = '%s'
+        chrs = '[\011%s]'
     else
         chrs = '['..escape(chrs)..']'
     end
@@ -478,7 +478,7 @@
         end
     until not new_equals
 
-    return equals 
+    return equals
 end
 
 --- Quote the given string and preserve any control or escape characters, such that reloading the string in Lua returns the same result.
@@ -499,8 +499,8 @@
         equal_signs = ("="):rep((equal_signs or -1) + 1)
         -- Long strings strip out leading newline. We want to retain that, when quoting.
         if s:find("^\n") then s = "\n" .. s end
-        local lbracket, rbracket =  
-            "[" .. equal_signs .. "[",  
+        local lbracket, rbracket =
+            "[" .. equal_signs .. "[",
             "]" .. equal_signs .. "]"
         s = lbracket .. s .. rbracket
     else
